airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java

```diff
@@ -77,14 +77,17 @@ public ConfigRepository(final ConfigPersistence persistence,
     this.database = new ExceptionWrappingDatabase(database);
   }
 
+  public ExceptionWrappingDatabase getDatabase() {
```

Comment:
required to run a query against the database.

Reply:
@davinchia this is breaking the whole point of the `ConfigRepository` abstraction. It also sets an example for everyone else who joins the team that the db can now be accessed from anywhere in code.

Is there a reason you can't write your query inside here instead like we do with other queries.

Reply:
My mental context at time of writing:
* I was under the impression we want to move away from this class to DatabaseConfigPersistence.java. Because of this, adding more code here didn't seem like a good idea.
* The queries were complicated multi-join queries and not the typical CRUD queries present in the ConfigRepository class. It didn't feel it was the right place to put these here.
* Chris mentioned it's useful to collect all the metric queries in one spot so he can re-use them for analytics if needed. A metric specific spot like [this class](https://github.com/airbytehq/airbyte/blob/master/airbyte-metrics/lib/src/main/java/io/airbyte/metrics/lib/MetricQueries.java) seemed more sane than adding to the ConfigRepository class which is bloated and quite tough to read at this point. (I think we should starting composing down to table-specific classes.)
* I feel that the accessing the db directly is 'okay' in special cases. In this case, the metric queries are quite specialised and unlikely to be used by another application. We don't want to use the normal CRUD operations here since it'll result in tens of queries vs 1.

I'm happy to move the queries here if you have feel otherwise.